aptos-move/aptos-vm/src/sharded_block_executor/block_executor_client.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright © Aptos Foundation
+
+use crate::block_executor::BlockAptosVM;
+use aptos_state_view::StateView;
+use aptos_types::transaction::{Transaction, TransactionOutput};
+use move_core_types::vm_status::VMStatus;
+use std::sync::Arc;
+
+pub trait TBlockExecutorClient {
+    fn execute_block<S: StateView + Sync>(
+        &self,
+        transactions: Vec<Transaction>,
+        state_view: &S,
+        concurrency_level: usize,
+        maybe_block_gas_limit: Option<u64>,
+    ) -> Result<Vec<TransactionOutput>, VMStatus>;
```

Comment:
Do we really need to return VMStatus here?

Reply:
Yes, otherwise, we would have to change the VM apis and that is going to be messy. This ensures that the client doesn't need to distinguish between local execution vs remote. What is the concern here?

Reply:
I'm confused.. if you mean you wanna use the client in place of `AptosVM`, why not just implement `VMExecutor` for the client? otherwise the caller side wouldn't require you to return VMStatus anyway.. Guess I'll need to read your next PR.

Reply:
@msmouse - Even if we implement `VMExecutor` for the client, the interface still requires it to return `VMStatus`? Also, IMO its not clean for the client to implement `VMExecutor` as the executor exposes both sharded and unsharded execution APIs, which this client can't expose.